Fileshare userguide (#11)
opensds/documentation: readthedocs/starterguides/userguides/fileshare_userguide.md

### Install fileshare feature with opensds
Follow the wiki link to install opensds https://github.com/opensds/opensds/wiki/OpenSDS-Cluster-Installation-through-Ansible

Make sure that backend nfs is enable to use fileshare feature 
###### NFS
If `nfs` is chosen as storage backend, modify `group_vars/osdsdock.yml`, native nfs driver dependent on lvm:
```yaml
enabled_backends: nfs
```

## Set the environment to start use of fileshare feature

### OpenSDS CLI
Firstly configure opensds CLI tool:
```bash
sudo cp /opt/opensds-hotpot-linux-amd64/bin/osdsctl /usr/local/bin/

export OPENSDS_ENDPOINT=http://{your_real_host_ip}:50040
export OPENSDS_AUTH_STRATEGY=keystone
export OS_AUTH_URL=http://{your_real_host_ip}/identity
export OS_USERNAME=admin
export OS_PASSWORD=opensds@123
export OS_TENANT_NAME=admin
export OS_PROJECT_NAME=admin
export OS_USER_DOMAIN_ID=default

osdsctl pool list # Check if the pool resource is available
```

#### Profile creation for fileshare
* Create profile for fileshare

```yaml

> osdsctl profile create '{"name":"test_profile", "description":"test policy for fileshare", "storageType":"file"}'

Reviewer: I think this command is duplicate in this doc, please have a check.

Author: Yes i removed duplication.

export OS_TENANT_NAME=admin
export OS_PROJECT_NAME=admin
export OS_USER_DOMAIN_ID=default

osdsctl pool list # Check if the pool resource is available
```

#### File share operations
Create a file share profile
```
osdsctl profile create '{"name":"Prf_fileshare", "description":"default policy for fileshare", "storageType":"file"}'
```

Create a file share
```
osdsctl fileshare create 1 -n <fileshare_name> -p <profile_id>
```
List all file shares

```
osdsctl fileshare list
```
Delete the file share
```
osdsctl fileshare delete <fileshare id>
```
Set access for file share

```
osdsctl fileshare acl create <fileshare_id> -a <client_ip> -c <list_of_access_capabilities> -t "ip"
```

Create snapshot of file share
```
osdsctl fileshare snapshot create <fileshare_id> -n <snapshot_name> -d "test snapshot"
```

### OpenSDS UI
OpenSDS UI dashboard is available at `http://{your_host_ip}:8088`, please login the dashboard using the default admin credentials: `admin/opensds@123`. Create `tenant`, `user`, and `profiles` as admin.

Logout of the dashboard as admin and login the dashboard again as a non-admin user to manage storage resource:

#### File Share Services
* Create a file share

> When you login to create file share is available in Resource tab->file share->create will allow you to create file share

Reviewer 2: Can we rephrase this line? 
```suggestion
Post login into the OpenSDS UI, File share can be created from: Resource -> File share -> Create
```

(Rephrase above line as per you)

> ![Fileshare create menu](https://github.com/Shruthi-1MN/documentation/blob/fileshare-userguide/readthedocs/starterguides/userguides/fileshare_menu.PNG)

Reviewer 2: Once this PR is merged, above URL will work?

* Create snapshot

Click on file share name, you can find create under snapshot menu, to create snapshot

> another quick option file share resource menu->file share name-> in same row you can find Create Snapshot link to create snapshot

Reviewer 2: ```suggestion
Can we break it into steps:
a) Click on File Share name
b) Under Snapshot menu, click on 'Create'

Also the quick option should be another sub-section here
```

> * Set access permission on file share (IP based access permissions are allowed)

Reviewer 2: What are access permissions? Why one need to set Access permissions? Can we add them?

Click on file share name, you can find create under access menu to create access
another quick option file share resource menu->file share name-> in same row you can find Create Access link to create access

![Fileshare snapshot and access](https://github.com/Shruthi-1MN/documentation/blob/fileshare-userguide/readthedocs/starterguides/userguides/snapshot_access.PNG)

### Some file share standards and limitations

> * File share with the same name is not allowed to create

Reviewer 2: ```suggestion
* File share with the same name cannot be create
```

> * File share should start with alphanumeric, special characters are not allowed

Reviewer 2: Break it into two line.
```suggestion
* File share can contain alphanumeric characters. Special characters are not allowed
```

> * Snapshot with snapshot name is not allowed to create

Reviewer 2: This line is vague. Could not understand it

* Access permission set is allowed only by client IP's

> * Put operations of ACL is not permitted for now

Reviewer 2: Please elaborate above two lines
